airflow/utils/operator_helpers.py

```diff
@@ -164,7 +164,10 @@ def determine(
 
     def unpacking(self) -> Mapping[str, Any]:
         """Dump the kwargs mapping to unpack with ``**`` in a function call."""
-        if self._wildcard and isinstance(self._kwargs, Context):
+        # Context is a TypedDict at lint time, and Mypy would complain it cannot
+        # be used in isinstance. But the call works at runtime since the type is
+        # actually implemented as a custom mapping, so we ignore the Mypy error.
+        if self._wildcard and isinstance(self._kwargs, Context):  # type: ignore
```

Comment:
Might be good also to review other type-ignores to see if we can silence a specific error (watch-out for `misc`)

Reply:
Have gone through all the ones added in this PR and converted them to specific codes where possible.